examples/nanocoap_server/coap_handler.c

```diff
@@ -166,14 +166,8 @@ ssize_t _sha256_handler(coap_pkt_t* pkt, uint8_t *buf, size_t len, coap_request_
     return pkt_pos - (uint8_t*)pkt->hdr;
 }
 
-/* must be sorted by path (ASCII order) */
-const coap_resource_t coap_resources[] = {
-    COAP_WELL_KNOWN_CORE_DEFAULT_HANDLER,
-    { "/echo/", COAP_GET | COAP_MATCH_SUBTREE, _echo_handler, NULL },
-    { "/riot/board", COAP_GET, _riot_board_handler, NULL },
-    { "/riot/value", COAP_GET | COAP_PUT | COAP_POST, _riot_value_handler, NULL },
-    { "/riot/ver", COAP_GET, _riot_block2_handler, NULL },
-    { "/sha256", COAP_POST, _sha256_handler, NULL },
-};
-
-const unsigned coap_resources_numof = ARRAY_SIZE(coap_resources);
+ADD_COAP_RESOURCE(_echo)   { "/echo/", COAP_GET | COAP_MATCH_SUBTREE, _echo_handler, NULL };
+ADD_COAP_RESOURCE(_board)  { "/riot/board", COAP_GET, _riot_board_handler, NULL };
+ADD_COAP_RESOURCE(_value)  { "/riot/value", COAP_GET | COAP_PUT | COAP_POST, _riot_value_handler, NULL };
+ADD_COAP_RESOURCE(_ver)    { "/riot/ver", COAP_GET, _riot_block2_handler, NULL };
+ADD_COAP_RESOURCE(_sha256) { "/sha256", COAP_POST, _sha256_handler, NULL };
```

Comment:
```suggestion
ADD_COAP_RESOURCE(echo)   { "/echo/", COAP_GET | COAP_MATCH_SUBTREE, _echo_handler, NULL };
ADD_COAP_RESOURCE(board)  { "/riot/board", COAP_GET, _riot_board_handler, NULL };
ADD_COAP_RESOURCE(value)  { "/riot/value", COAP_GET | COAP_PUT | COAP_POST, _riot_value_handler, NULL };
ADD_COAP_RESOURCE(ver)    { "/riot/ver", COAP_GET, _riot_block2_handler, NULL };
ADD_COAP_RESOURCE(sha256) { "/sha256", COAP_POST, _sha256_handler, NULL };
```

We could prepend a prefix to the names in the `ADD_COAP_RESOURCE()` macro to avoid name clashes. Then it should be safe to no longer have a leading underscore. (Also, warnings about reserved names tend to be annoying...)